server/utils/customErrors.js

```diff
@@ -1,3 +1,11 @@
+/**
+ * GY: This is a good pattern to use. 
+ * - I recently adopted this as well to create different types of errors
+ *   - internal errors: include a lot more debug information, but shouldn't
+ *     expose by API
+ *   - external errors: converting internal errors to externally visible one
+ *     - i.e stripping out sensitive data, wrapping with better error message
+ */
```

Comment:
Thank you for the suggestion, I got it! I will try to create another internal error class for debugging purposes as well (currently this custom error class only contains an error message to expose by API).

Reply:
To be honest, this is not a high priority at the moment. But eventually you'll see examples of this show up in bigger projects.